lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,24 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n*k), where n is the size of the external array that represents
+# the rows and k is the size of an individual array that represents the columns
```

Comment:
[nit] Since the problem statement mandates square matrices, `n` and `k` should be equal.

Reply:
Also, technically, your implementation is O(n) because it only looks at the first row & column. 😉